src/Models/Apis.ts

```diff
@@ -3,27 +3,27 @@
 
 import * as vscode from "vscode";
 
-import { AzureAccount } from "../azure-account.api";
-import { AzureAccountCommands } from "../common/Commands";
+import { AzureAccountCommands, VscodeCommands } from "../common/Commands";
 import { DependentExtensionNotFoundError } from "../common/Error/OperationFailedErrors/DependentExtensionNotFoundError";
-
-import { ExtensionName } from "./Interfaces/Api";
+import { ExtensionName, ExtensionNameIdMap } from "./Interfaces/Api";
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export function getExtension(name: ExtensionName): any {
-  switch (name) {
-    case ExtensionName.Toolkit: {
-      const toolkit = vscode.extensions.getExtension("vsciot-vscode.azure-iot-toolkit");
-      return toolkit ? toolkit.exports : undefined;
-    }
-    case ExtensionName.AzureAccount: {
-      const azureAccount = vscode.extensions.getExtension<AzureAccount>("ms-vscode.azure-account");
-      return azureAccount ? azureAccount.exports : undefined;
-    }
-    default: {
-      return undefined;
+  const extensionId = ExtensionNameIdMap.get(name);
+  if (extensionId) {
+    const extension = vscode.extensions.getExtension(extensionId);
+    if (extension) {

```

Comment:
It seems that it is not necessary to list all enum, since they the same.
``` return extension && entension.exports ? extension.exports : undefined;
```